src/nvim/ex_docmd.c

```diff
@@ -9411,3 +9411,36 @@ static void ex_terminal(exarg_T *eap)
     xfree(name);
   }
 }
+
+static void ex_plug(exarg_T *eap)
+{
+  char *path = NULL;
+  if (*eap->arg != NUL) {
+    if (os_isdir(eap->arg)) {  // full path: plug ~/.config/nvim/plugged/coolplugin
+      path = (char *)vim_strsave_fnameescape(eap->arg, FALSE);
+    } else {  // spec: plug neovim/neovimplugin
+      char_u *user_vim = vim_strsave(p_rtp);
+      strtok((char *)user_vim, ","); // split(&runtimepath, ',')[0]
+      char_u *basename = path_tail((char_u *)eap->arg); // 'neovim/neovimplugin' -> 'neovimplugin'
+      if (STRNCMP(basename, "", 1) != 0 && STRNCMP(p_bundledir, "", 1) != 0) {
+        if (!os_isdir(p_bundledir)) {
+          path = concat_fnames(
+              concat_fnames((char *)user_vim,
+                (char *)p_bundledir, true),
+              (char *)basename, true);
+        } else {
+          path = concat_fnames((char *)p_bundledir, (char *)basename, true);
+        }
+      }
+    }
+
+    if (path != NUL) {
+      char ex_cmd[512];
+      snprintf(ex_cmd, sizeof(ex_cmd), ":set rtp+=%s", path);
```

Comment:
`path` is unescaped. If you use this code you must escape comma in path, and also all special characters.